nodejs/eks/cluster.ts

```diff
@@ -1064,6 +1009,75 @@ export function createCore(
             return result;
         });
 
+    const corednsExplicitlyEnabled =
+        args.corednsAddonOptions?.enabled === true || args.corednsAddonOptions?.configurationValues;
+    // We can only enable the coredns addon if we have a node group to place it on
+    // This means we are either using the default node group or the cluster is a fargate cluster
+    // Also, if the user explicitly enables it then do what they want
+    pulumi.output(args.fargate).apply((fargate) => {
+        if (
+            corednsExplicitlyEnabled ||
+            ((fargate || !args.skipDefaultNodeGroup) && (args.corednsAddonOptions?.enabled ?? true))
+        ) {
+            const corednsVersion: pulumi.Output<string> = args.corednsAddonOptions?.version
+                ? pulumi.output(args.corednsAddonOptions.version)
+                : aws.eks
+                      .getAddonVersionOutput(
+                          {
+                              addonName: "coredns",
+                              kubernetesVersion: eksCluster.version,
+                              mostRecent: true, // whether to return the default version or the most recent version for the specified kubernetes version
+                          },
+                          { parent, provider },
+                      )
+                      .apply((addonVersion) => addonVersion.version);
+
+            const configurationValues = pulumi
+                .all([args.fargate, args.corednsAddonOptions?.configurationValues])
+                .apply(([fargate, configurationValues]) => {
+                    if (fargate) {
+                        return {
+                            computeType: "Fargate",
+                            ...configurationValues,
+                        };
+                    } else {
+                        return configurationValues;
+                    }
+                });
+
+            const corednsAddon = new aws.eks.Addon(
+                `${name}-coredns`,
+                {
+                    clusterName: eksCluster.name,
+                    addonName: "coredns",
+                    tags: args.tags,
+                    preserve: true,
+                    addonVersion: corednsVersion,
+                    resolveConflictsOnCreate:
+                        args.corednsAddonOptions?.resolveConflictsOnCreate ?? "OVERWRITE",
+                    resolveConflictsOnUpdate:
+                        args.corednsAddonOptions?.resolveConflictsOnUpdate ?? "OVERWRITE",
+                    configurationValues: stringifyAddonConfiguration(configurationValues),
+                },
+                {
+                    parent,
+                    provider,
+                    dependsOn: fargateProfile.apply((profile) => {
```

Comment:
That's the fix for the Fargate issues. I needed to include it here otherwise I couldn't test the changes in this PR.